### PR TITLE
feat: Franprix

### PR DESCRIPTION
**Pull request by Romeo-PHILLIPS**
857 stores

### CODE REVIEW

**Review by Cj-Malone — APPROVED** [review]
Can confirm this works from a French IP.

```
{'atp/field/email/missing': 857,
 'atp/field/image/missing': 857,
 'atp/field/opening_hours/missing': 14,
 'atp/field/phone/invalid': 4,
 'atp/field/phone/missing': 800,
 'atp/field/state/missing': 857,
 'atp/field/twitter/missing': 857,
 'atp/nsi/perfect_match': 857,
 'downloader/request_bytes': 979,
 'downloader/request_count': 2,
 'downloader/request_method_count/GET': 2,
 'downloader/response_bytes': 107308,
 'downloader/response_count': 2,
 'downloader/response_status_count/200': 2,
 'elapsed_time_seconds': 6.460295,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2022, 11, 1, 16, 42, 49, 758814),
 'httpcompression/response_bytes': 1295431,
 'httpcompression/response_count': 2,
 'item_scraped_count': 857,
 'log_count/DEBUG': 868,
 'log_count/INFO': 8,
 'log_count/WARNING': 1,
 'memusage/max': 81149952,
 'memusage/startup': 81149952,
 'response_received_count': 2,
 'robotstxt/request_count': 1,
 'robotstxt/response_count': 1,
 'robotstxt/response_status_count/200': 1,
 'scheduler/dequeued': 1,
 'scheduler/dequeued/memory': 1,
 'scheduler/enqueued': 1,
 'scheduler/enqueued/memory': 1,
 'start_time': datetime.datetime(2022, 11, 1, 16, 42, 43, 298519)}
```

**Comment by Romeo-PHILLIPS** [comment]
Would be cool that one day we can specify a country IP based proxy to run the Spider

**Comment by Cj-Malone** [comment]
Yeah absolutely, there was talk about [this](https://www.zyte.com/smart-proxy-manager/) recently but @Scrapinghub refused to give us an Open Source discount. There is a few GB spiders that work locally but not on the weekly, I still want to see it done.

**Comment by mjoe999** [comment]
If you change the spider to be "franprix_fr.py" and name = "franprix_fr" then you can drop the country from the "item_attributes" as pipeline code will now add this for you. It also makes it easy in the future for us to pick up on "*_fr.py" or the spider name as a strong hint to run the spider against a French proxy.

**Comment by Romeo-PHILLIPS** [comment]
> If you change the spider to be "franprix_fr.py" and name = "franprix_fr" then you can drop the country from the "item_attributes" as pipeline code will now add this for you. It also makes it easy in the future for us to pick up on "*_fr.py" or the spider name as a strong hint to run the spider against a French proxy.

Oh okay thank you I didn't know that

**Comment by mjoe999** [comment]
The pipeline code was only added a few days ago :-) I also hope you got my reply concerning your question on:
https://www.carhartt-wip.com/fr/stores/4977

**Comment by Romeo-PHILLIPS** [comment]
> The pipeline code was only added a few days ago :-) I also hope you got my reply concerning your question on: https://www.carhartt-wip.com/fr/stores/4977

Yea I did see your response thank you! I had deleted my original question thinking maybe the website had some client side rendering thats why it wasn't being parsed correctly inside Scrapy. Hopefully we can make it even better for the future :-)

File: locations/spiders/franprix_fr.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    open_time=formatted_hours[0],
+                    close_time=formatted_hours[1],
+                    time_format="%H:%M",
+                )
+            properties = {
+                "ref": store.get("id"),
+                "name": f"Franprix - {store.get('store_name')}"
+                if store.get("store_name")
+                else "Franprix",
+                "street_address": store.get("street"),
```
Comment (Cj-Malone) [inline locations/spiders/franprix_fr.py:41]: `street` has city and postcode, so I'd have put it in `addr_full`, not `street_address`.

**Review by Cj-Malone — APPROVED** [review]

**Pull request merged**